packages/lib/date-ranges.ts

```diff
@@ -158,7 +178,16 @@ export function buildDateRanges({
     (ranges) => ranges.filter((range) => range.start.valueOf() !== range.end.valueOf())
   );
 
-  return dateRanges.flat();
+  const dateRangesWithoutOOO = Object.values({
+    ...groupedWorkingHours,
+    ...groupedDateOverrides,
+    ...groupedOOO,
+  }).map(
+    // remove 0-length overrides && OOO dates that were kept to cancel out working dates until now.
+    (ranges) => ranges.filter((range) => range.start.valueOf() !== range.end.valueOf())
```

Comment:
OOO always starts and end on the same time (handled the same way as a full-day date override)